cpp/util/etcd.cc

```diff
@@ -1038,19 +1023,9 @@ void EtcdClient::Watch(const string& key, const WatchCallback& cb,
 }
 
 
-static void GenericAdapter(util::Status status,
-                           const std::shared_ptr<JsonObject>& json_body,
-                           int64_t etcd_index,
-                           EtcdClient::GenericResponse* resp, Task* task) {
-  resp->etcd_index = etcd_index;
-  resp->json_body = json_body;
-  task->Return(status);
-}
-
-
 void EtcdClient::Generic(const string& key, const map<string, string>& params,
-                         evhttp_cmd_type verb, GenericResponse* resp,
-                         Task* task) {
+                         evhttp_cmd_type verb, bool separate_conn,
```

Comment:
`separate_conn` might be nicer as an `enum` for readability at call sites, unless, you know, this just happens to be going away RSN :)